src/deploy/osp_deployer/profile.py

```diff
@@ -168,3 +175,23 @@ def valid_buffer_pool_size(self, value):
             if val < 0:
                 return False
         return True
+
+    def valid_nic_env_file(self, config_file):
+        try:
+            if self.settings.enable_sriov and self.settings.enable_ovs_dpdk:
+                assert config_file == "ovs-dpdk_sriov_9_port/nic_environment.yaml"
+            elif self.settings.enable_sriov:
+                assert config_file == "sriov_7_port/nic_environment.yaml" or \
+                    config_file == "sriov_9_port/nic_environment.yaml"
+            elif self.settings.enable_ovs_dpdk:
+                assert config_file == "ovs-dpdk_7_port/nic_environment.yaml" or \
+                    config_file == "ovs-dpdk_9_port/nic_environment.yaml"
+            elif self.settings.enable_dvr:
+                assert config_file == "5_port/nic_environment.yaml" or \
+                    config_file == "dvr_7_port/nic_environment.yaml"    
+            else:
+                assert config_file == "5_port/nic_environment.yaml" or \
+                    config_file == "4_port/nic_environment.yaml"
+            return True
+        except AssertionError:
+            return False
```

Comment:
This sort of defeats the intent of the nic env file implementation ..i think i 'd happy with just having a list of allowed nic's files in the profile definition and not the above.
We're trying to keep all the profile related information outside of the codes and contained within the profile definition. The above would require maintenance every time we add/remove nic files.
@cdearborn any opinion ?

Reply:
I agree.  We should not be hard coding the names of the NIC env files.  This would force code changes every time we wanted to add or remove NIC env files.